egs/cfg2/wip/kaldi_grammar.py

```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/env python
+import os
+from grammar_parser.cfgparser import CFGParser
+
+
+class KaldiGrammar:
+    """
+    Class KaldiGrammar uses as input a grammar file with extension '.fcfg' and has two functions:
+    get_rule_element: extracts the defined grammar rules
+    get_words: extracts the unique words and creates 'corpus.txt' which is used to build 'G.fst'
+    """
+    def __init__(self, grammar_file_string, target):
+
+        if os.path.exists(grammar_file_string):
+            self.parser = CFGParser.fromfile(grammar_file_string)
+            self.grammar_file = grammar_file_string
+        else:
+            self.parser = CFGParser.fromstring(grammar_file_string)
+            self.grammar_string = grammar_file_string
+
+        self.target = target
+
+    def get_rule_element(self, depth=0):
+        """
+        Extracts the grammar rules, defined within the grammar file
+
+        :param depth: level of depth of the grammar rule
+        :return: grammar tree
+        """
+        RulesList = {}
+        # print(depth)
+        if self.target not in self.parser.rules:
```

Comment:
Remove self from targets throughout the function

Reply:
done